autoload/dbext_dbi.vim

```diff
@@ -1539,7 +1695,52 @@ sub db_print_results
             }
         }
     }
+    db_debug("db_print_results: before last_line(...rows...), last_line=$last_line");
     db_vim_print($last_line, "(".scalar(@result_set)." rows)");
+    if (@dbms_output) {
+      $last_line++;
+      # ans: we need this "join and afterwards split" trick because otherwise
+      # individual rows from @dbms_output are coming encoded differently, see as well
+      # exploit Perl_Oracle_DBD_dbms_output/perl_dbd_oracle_dbms_output__inside_vim__simple.vim
+      # call :so % | call Check_my_perl__load()
+      ##my @dbms_output_remade = split ("\n", join("\n", @dbms_output));
+      # ans: 2021-09-19 now we use decode("utf8"...) and other method of
+      # fetching of array, there are chances we dont need  this ugly workaround
+      # any more
+      db_debug("db_print_results: \@dbms_output size: ".scalar(@dbms_output));
+      my @dbms_output_remade = @dbms_output;
+      for my $dol0 (@dbms_output_remade) {
+        #### THIS IS MAGIC !!!!!!!
+        #### THIS_IS_MAGIC !!!!!!!
+        #### THIS IS MAGIC !!!!!!!
```

Comment:
What is magic?

Reply:
I was too excited of being able to figure out, that ```decode('utf8', $str)``` solves the problem with dbms_output non-ASCII chars being wrongly encoded ))) Cleaned up this comments a bit as well as extracted example referenced which reproduces a problem as separate file on github (hope it is available publicly)